SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -185,20 +188,13 @@ private void submitRequest(SingularityRequest request, Optional<SingularityReque
         });
 
         if (request.isRackSensitive() && configuration.isRebalanceRacksOnScaleDown()) {
-          List<SingularityTaskId> extraCleanedTasks = new ArrayList<>();
-          int numActiveRacks = rackManager.getNumActive();
-          double perRack = request.getInstancesSafe() / (double) numActiveRacks;
-
-          Multiset<String> countPerRack = HashMultiset.create();
-          for (SingularityTaskId taskId : remainingActiveTasks) {
-            countPerRack.add(taskId.getSanitizedRackId());
-            LOG.info("{} - {} - {} - {}", countPerRack, perRack, extraCleanedTasks.size(), taskId);
-            if (countPerRack.count(taskId.getSanitizedRackId()) > perRack && extraCleanedTasks.size() < numActiveRacks / 2) {
-              extraCleanedTasks.add(taskId);
-              LOG.info("Cleaning up task {} to evenly distribute tasks among racks", taskId);
-              taskManager.createTaskCleanup(new SingularityTaskCleanup(user.getEmail(), TaskCleanupType.REBALANCE_RACKS, System.currentTimeMillis(), taskId, Optional.<String>absent(), Optional.<String>absent(), Optional.absent()));
-            }
-          }
+          rebalancingHelper.rebalanceRacks(request, remainingActiveTasks, user.getEmail());
+          // Why not remove cleaned tasks from remainingActiveTasks here like in SingualrityScheduler?
+          // Why does scheduler do straight up user, and this does user.getemail?
```

Comment:
I believe the username is often the email anyways. It can be the username instead there though, that just powers what shows up in the UI (e.g. user that triggered the cleanup).

`remainingActiveTasks` here already excludes those that were cleaned up. In the loop above this we are only adding to that list if we don't create a cleanup